Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -329,6 +329,24 @@ void UPSWorldSubsystem::OnTakeActorsFromPoolCompleted(const TArray<FPoolObjectDa
 	}
 }
 
+// Returns spot component by the player tag, returns null if spot is not found
+UPSSpotComponent* UPSWorldSubsystem::GetSpotComponentByPlayerTag(FPlayerTag PlayerTag)
+{
+	if (PSSpotComponentArrayInternal.IsEmpty())
+	{
+		return nullptr;
+	}
+	
+	for (UPSSpotComponent* SpotComponent : PSSpotComponentArrayInternal)
+	{
+		if (SpotComponent->GetMeshChecked().GetPlayerTag() == PlayerTag)
```

Comment:
`SpotComponent` might be null (garbage collected)
Do `if (SpotComponent && SpotComponent->`

Reply:
added https://github.com/h4rdmol/ProgressionSystem/pull/40/commits/4155fb5bd7315af3cc5ac3d3f35fb75161d73660